### PR TITLE
Update mandatory schema metadata validation

### PR DESCRIPTION
**Pull request by petechd**
### PR Context
Adds more validation rules to schema mandatory metadata validation based on different themes and their usage in runner templates.

### Checklist

* [ ] eq-translations updated to support any new schema keys which need translation

### CODE REVIEW

**Comment by petechd** [comment]
[Update mandatory schema metadata validation (2 SP)](https://trello.com/c/AfanXCWD/5367-update-mandatory-schema-metadata-validation-2-sp)

File: app/validators/metadata_validator.py [hunk 1]
```diff
@@ -29,11 +30,19 @@ def validate_duplicates(self):
...
+
+        if self.theme_name in ["business", "default", "northernireland"]:
+            required_metadata_names.extend(["user_id", "period_id", "ru_name"])
+        elif self.theme_name in ["census", "census-nisra"]:
+            if self.form_type:
+                required_metadata_names.extend(
+                    ["user_id", "period_id", "display_address"]
+                )
+            else:
+                required_metadata_names.extend(["user_id", "period_id"])
```
Comment (MebinAbraham) [inline app/validators/metadata_validator.py:45]: I wouldn't bother with census theme. They need revisiting if we ever needed to use it again.

File: app/validators/metadata_validator.py [hunk 2]
```diff
@@ -29,11 +30,19 @@ def validate_duplicates(self):
...
-        required_metadata_names = ["user_id", "period_id"]
-
-        if self.theme_name in ["default", "northernireland"]:
-            required_metadata_names.append("ru_name")
+        # ru_name required for template rendering in business, default and NI theme
+        # display_address required for template rendering in census, census-nisra theme when form_type exists
+        required_metadata_names = []
+
+        if self.theme_name in ["business", "default", "northernireland"]:
+            required_metadata_names.extend(["user_id", "period_id", "ru_name"])
```
Comment (MebinAbraham) [inline app/validators/metadata_validator.py:38]: Once RAS moves to v2 format, we can remove `user_id` and `period_id`.

**Review by MebinAbraham — APPROVED** [review]

File: app/validators/metadata_validator.py (hunk 1 above)
Comment (petechd) [reply app/validators/metadata_validator.py]: Removed!

File: app/validators/metadata_validator.py (hunk 2 above)
Comment (petechd) [reply app/validators/metadata_validator.py:37]: Ok, I will keep it for now!

File: app/validators/metadata_validator.py [hunk 3]
```diff
@@ -29,11 +29,12 @@ def validate_duplicates(self):
 
     def validate_mandatory(self):
         # user_id and period_id required downstream for receipting
-        # ru_name required for template rendering in default and NI theme
-        required_metadata_names = ["user_id", "period_id"]
+        # ru_name required for template rendering in business, default and NI theme
+        # display_address required for template rendering in census, census-nisra theme when form_type exists
```
Comment (berroar) [inline app/validators/metadata_validator.py:33]: This comment can be removed 👍

Comment (petechd) [reply app/validators/metadata_validator.py]: Removed!

**Review by berroar — APPROVED** [review]

**Pull request merged**